src/h5web/App.test.tsx

```diff
@@ -1,23 +1,134 @@
+
+  const noVisText1 = queryByText('Nothing to visualize');
+  expect(noVisText1).not.toBeInTheDocument();
+
+  const groupIdRow1 = await findByRole('row', { name: /Entity ID/ });
+  expect(groupIdRow1).toBeVisible();
+
+  // Switch back to "display" mode
+  fireEvent.click(displayBtn);
+
+  const groupIdRow2 = queryByRole('row', { name: /Entity ID/ });
+  expect(groupIdRow2).not.toBeInTheDocument();
+
+  const noVisText2 = await findByText('Nothing to visualize');
+  expect(noVisText2).toBeVisible();
+});
+
+test('toggle explorer sidebar', async () => {
+  const { findByRole, getByRole } = renderApp();
+
+  const domainBtn = await findByRole('treeitem', { name: DOMAIN });
+  const sidebarBtn = getByRole('button', {
+    name: 'Toggle explorer sidebar',
+  });
+
+  expect(domainBtn).toBeVisible();
+  expect(sidebarBtn).toHaveAttribute('aria-pressed', 'true');
+
+  fireEvent.click(sidebarBtn);
+  expect(domainBtn).not.toBeVisible();
+  expect(sidebarBtn).toHaveAttribute('aria-pressed', 'false');
+
+  fireEvent.click(sidebarBtn);
+  expect(domainBtn).toBeVisible();
+  expect(sidebarBtn).toHaveAttribute('aria-pressed', 'true');
+});
+
+test('navigate groups in explorer', async () => {
+  const { findByRole, queryByRole, getByRole } = renderApp();
+
+  const groupBtn = await findByRole('treeitem', { name: 'entry_0000' });
+  expect(groupBtn).toHaveAttribute('aria-selected', 'false');
+  expect(groupBtn).toHaveAttribute('aria-expanded', 'false');
+
+  // Expand `entry_0000` group
+  fireEvent.click(groupBtn);
+
+  expect(groupBtn).toHaveAttribute('aria-selected', 'true');
+  expect(groupBtn).toHaveAttribute('aria-expanded', 'true');
+
+  const childGroupBtn = getByRole('treeitem', { name: '1_cormap' });
+  expect(childGroupBtn).toHaveAttribute('aria-selected', 'false');
+  expect(childGroupBtn).toHaveAttribute('aria-expanded', 'false');
+
+  // Expand `1_cormap` child group
+  fireEvent.click(childGroupBtn);
+
+  expect(groupBtn).toHaveAttribute('aria-selected', 'false');
+  expect(groupBtn).toHaveAttribute('aria-expanded', 'true');
+  expect(childGroupBtn).toHaveAttribute('aria-selected', 'true');
+  expect(childGroupBtn).toHaveAttribute('aria-expanded', 'true');
+
+  // Collapse `1_cormap` child group
+  fireEvent.click(childGroupBtn);
+
+  expect(childGroupBtn).toHaveAttribute('aria-selected', 'true');
+  expect(childGroupBtn).toHaveAttribute('aria-expanded', 'false');
+
+  // Select `entry_0000` group
+  fireEvent.click(groupBtn);
+
+  expect(groupBtn).toHaveAttribute('aria-selected', 'true');
+  expect(groupBtn).toHaveAttribute('aria-expanded', 'true'); // remains expanded as it wasn't previously selected
+  expect(childGroupBtn).toHaveAttribute('aria-selected', 'false');
+
+  // Collapse `entry_0000` group
+  fireEvent.click(groupBtn);
+
+  expect(queryByRole('treeitem', { name: '1_cormap' })).not.toBeInTheDocument();
+
+  // `waitFor` avoids warning due to async work occuring after completion of test
+  await waitFor(() => {
```

Comment:
If something asynchronous happens after performing an action (with `fireEvent.click()` for instance), you have three options:

- if you know an element will be added to the DOM, use `await findBy*(...)`;
- if you know an element will be removed from the DOM, use `await waitForElementToBeRemoved(() => queryBy*(...))`
- in all other cases, use `await waitFor(() => { /* something that throws until the async operation has completed, like `expect or `getBy` */ })`

In this particular case, as the comment states, `waitFor` is only required because some async work that I don't really care about occurs.